nima/http/media/jackson/src/main/java/io/helidon/nima/http/media/jackson/JacksonMediaSupportProvider.java

```diff
@@ -0,0 +1,147 @@
+        objectMapper = new ObjectMapper()
+                .registerModule(new ParameterNamesModule())
+                .registerModule(new Jdk8Module())
+                .registerModule(new JavaTimeModule());
+
+        reader = new JacksonReader(objectMapper);
+        writer = new JacksonWriter(objectMapper);
+    }
+
+    @Override
+    public <T> ReaderResponse<T> reader(GenericType<T> type, Headers requestHeaders) {
+        if (requestHeaders.contentType()
+                .map(it -> it.test(MediaTypes.APPLICATION_JSON))
+                .orElse(true)) {
+            if (objectMapper.canDeserialize(objectMapper.constructType(type.type()))) {
+                return new ReaderResponse<>(SupportLevel.COMPATIBLE, this::reader);
+            }
+        }
+
+        return ReaderResponse.unsupported();
+    }
+
+    @Override
+    public <T> WriterResponse<T> writer(GenericType<T> type,
+                                        Headers requestHeaders,
+                                        WritableHeaders<?> responseHeaders) {
+        // check if accepted
+        for (HttpMediaType acceptedType : requestHeaders.acceptedTypes()) {
+            if (acceptedType.test(MediaTypes.APPLICATION_JSON)) {
+                if (objectMapper.canSerialize(type.rawType())) {
+                    return new WriterResponse<>(SupportLevel.COMPATIBLE, this::writer);
+                }
+                return WriterResponse.unsupported();
+            }
+        }
+
+        if (requestHeaders.acceptedTypes().isEmpty()) {
+            if (objectMapper.canSerialize(type.rawType())) {
+                return new WriterResponse<>(SupportLevel.COMPATIBLE, this::writer);
+            }
+        }
+
+        return WriterResponse.unsupported();
+    }
+
+    @Override
+    public <T> ReaderResponse<T> reader(GenericType<T> type,
+                                        Headers requestHeaders,
+                                        Headers responseHeaders) {
+        // check if accepted
+        for (HttpMediaType acceptedType : requestHeaders.acceptedTypes()) {
+            if (acceptedType.test(MediaTypes.APPLICATION_JSON) || acceptedType.mediaType().isWildcardType()) {
+                if (objectMapper.canDeserialize(objectMapper.constructType(type.type()))) {
+                    return new ReaderResponse<>(SupportLevel.COMPATIBLE, this::reader);
+                }
+            }
+        }
+
+        if (requestHeaders.acceptedTypes().isEmpty()) {
+            if (objectMapper.canDeserialize(objectMapper.constructType(type.type()))) {
+                return new ReaderResponse<>(SupportLevel.COMPATIBLE, this::reader);
+            }
+        }
+
+        return ReaderResponse.unsupported();
+    }
+
+    @Override
+    public <T> WriterResponse<T> writer(GenericType<T> type, WritableHeaders<?> requestHeaders) {
+        if (requestHeaders.contains(Http.Header.CONTENT_TYPE)) {
+            if (requestHeaders.contains(CONTENT_TYPE_JSON)) {
```

Comment:
I'm not familiar with all the rules and responsibilities of a `MediaProvider` but do bear in mind Jackson can read/write everything from JSON to properties to YAML. Naïvely this check looks to be too restrictive.